WordPress/Classes/Models/WPAccount.m

```diff
@@ -56,6 +56,27 @@ - (void)didTurnIntoFault
 
 #pragma mark - Custom accessors
 
+- (void)setUsername:(NSString *)username
+{
+    NSString *previousUsername = self.username;
+    
+    BOOL usernameChanged = ![previousUsername isEqualToString:username];
+    NSString *authToken = nil;
+    
+    if (usernameChanged) {
+        authToken = self.authToken;
+        self.authToken = nil;
+    }
+    
+    [self willChangeValueForKey:@"username"];
```

Comment:
I didn't want to change the behavior of how the setter operates - willChange, setPrimitive, didChange - is essentially what the dynamic code generator does.